src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/MiscellaneousProjectSnapshot.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.CodeAnalysis.Host;
+
+namespace Microsoft.CodeAnalysis.Razor.ProjectSystem
+{
+    internal class MiscellaneousProjectSnapshot : ProjectSnapshot
```

Comment:
Does this duplicate `EphemeralProjectSnapshot`? what do you see as different between these types?

Reply:
Only difference is the lacking file path. I made a separate type mostly for clarity in debug purposes. Aka, easy to differentiate projectless documents from "being initialized" project documents

Reply:
What would it look like to unify them? Do you think it's worth a separate concept? Would this look like a good decision in hindsight?

Reply:
I still don't understand how these differ. These projects still have the same lifetime.